arm-software/linux/2-getting_started.atfl.md

```diff
@@ -0,0 +1,317 @@
+
+### Increase the optimization level
+
+To control the optimization level, specify the `-O<level>` option on your
+compile line, and replace `<level>` with one of `0`, `1`, `2` or `3`. The `-O0`
+option is the lowest, and the default, optimization level. `-O3` is the highest
+optimization level. Arm compilers performs auto-vectorization at level `-O2` and
+above.
+
+For example, to compile the `source.c` file into a binary called `binary`, and
+use the `-O3` optimization level, use:
+
+```
+$ armclang -O3 -o binary source.c
+```
+
+To compile the `source.f90` file into a binary called binary, and use the `-O3`
+optimization level, use:
+
+```
+$ armflang -O3 -o binary source.f90
+```
+
+#### Note
+
+Similarly to other compilers, the Arm Toolchain for Linux C and C++ compilers
+can also be supplied with the `-Ofast option`. This will however result in
+displaying the following deprecation warning:
+
+```
+warning: argument '-Ofast' is deprecated; use '-O3 -ffast-math' for the same behavior, or '-O3' to enable only conforming optimizations [-Wdeprecated-ofast]
+```
+
+As the warning message states, the effect of applying the `-Ofast` option when
+compiling the C/C++ programs can be achieved by using the `-O3 -ffast-math`
+options instead.
+
+In case of Fortran, the use of the `-Ofast` option triggers the following
+deprecation warning:
+
+```
+warning: argument '-Ofast' is deprecated; use '-O3 -ffast-math -fstack-arrays' for the same behavior, or '-O3 -fstack-arrays' to enable only conforming optimizations [-Wdeprecated-ofast]
+```
+
+As the warning message states, the effect of applying the `-Ofast` option when
+compiling Fortram programs can be achieved by using the
+`-O3 -ffast-math -fstack-arrays` options instead.
+
+### Compile and optimize using CPU auto-detection
+
+If you tell the compiler what target CPU your application will run on, it can
+make target-specific optimization decisions. Target-specific optimization
+decisions help ensure your application runs as efficiently as possible. To tell
+the compiler to make target-specific compilation decisions, use the
+`-mcpu=<target>` option and replace `<target>` with your target processor (from
+a supported list of targets).
+
+In addition, the `-mcpu` option also supports the `native` argument.
+`-mcpu=native` enables the compiler to auto-detect the architecture and
+processor type of the CPU that you are running the compiler on.
+
+For example, to auto-detect the target CPU and optimize your C application for
+this target, use:
+
+```
+$ armclang -O3 -mcpu=native -o binary source.c
+```
+
+To auto-detect the target CPU and optimize your Fortran application for this
+target, use:
+
+```
+$ armflang -O3 -mcpu=native -o binary source.f90
+```
+
+The `-mcpu` option supports a range of Armv8-A-based Systems-on-Chips (SoCs).
+When `-mcpu` is not specified, by default, `-mcpu=generic` is set, which
+generates  portable output suitable for any Armv8-A-based target.
+
```

Comment:
Please add links to Clang and Flang commandline reference guides.

Reply:
https://releases.llvm.org/20.1.0/tools/clang/docs/ClangCommandLineReference.html
https://releases.llvm.org/20.1.0/tools/flang/docs/FlangCommandLineReference.html